spec/bolt/inventory_spec.rb

```diff
@@ -306,4 +192,23 @@ def targets(names)
       end
     end
   end
+
+  # TODO: Why do these fail?
```

Comment:
The way you've implemented `set_var` and `vars` doesn't create methods on the Bolt::Target Ruby class, only Puppet functions that bind to the Target datatype. So you would have to run those calls through the Puppet parser to do the kind of tests you have.

I would focus on testing the new Puppet functions in boltlib, and the new functions on Bolt::Inventory here.